Olapic/DevKit-ES6
# Update gulp-bundlerify, add esdoc-uploader and fix the distribution

#### What does this PR do?

- Updates [gulp-bundlerify](https://github.com/homer0/gulp-bundlerify) to `v1.0.3`.
 - Remove [Jest](https://github.com/facebook/jest/) from the dependencies, since `gulp-bundlerify` handles the tests now.
 - Instead of overwriting the tasks in order to uglify just for `build`, the `gulpfile` now uses `beforeTask`.
 - The [ESDoc](https://esdoc.org) options are read from the `esdoc.json` file, which is also used for the _documentation deployment_.
 - Added a `es5` task that builds the library for ES5 so it can be consumed from `npm` as it should.
- Adds [esdoc-uploader](https://github.com/homer0/esdoc-uploader) to make the _documentation deployment_ automatic.
- Fix the tests so they'll work on the Jest latest version.

#### How should this be manually tested?

1. Run all the project tasks (`npm run` to list them).
2. Try to consume it as a `node_module`, everything should work just fine.

#### Any background context you want to provide?

- The `main` property of the `package.json` now points to the ES5 compiled version because publishing code that is not ready to be _required_ it's considered an anti-pattern.
- You may noticed the the coverage percents for statements and branchs are WAY lowers that on the initial version, well, the reason is that the last version of [babel-jest](https://github.com/babel/babel-jest) removed `auxiliaryCommentBefore`, which tells [Istanbul](https://github.com/gotwarlost/istanbul/) to skip the extra lines that Babel adds. Also, there seems to be an issue with coverage and the ES6 `import` syntax, which also lowers the percentage.
